orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/tasks/WaitTask.groovy

```diff
@@ -43,7 +43,9 @@ class WaitTask implements RetryableTask {
     def waitTimeMs = TimeUnit.MILLISECONDS.convert(waitTime, TimeUnit.SECONDS)
     def now = timeProvider.millis
 
-    if (!stage.context.containsKey("waitTaskState") || !stage.context.waitTaskState instanceof Map) {
+
+    def waitTaskState = stage.context.waitTaskState
+    if (!waitTaskState || !waitTaskState instanceof Map) {
```

Comment:
`waitTaskState` can now be an empty map so checking not null and not empty.

(would be slick if we had a consistent pattern for removing keys from the context, perhaps another attribute of the `TaskResult`)